test/WebSites/ModelBindingWebSite/Controllers/BindModelController.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc;
+using ModelBindingWebSite.Models;
+
+namespace ModelBindingWebSite.Controllers
+{
+    public class BindModelController : Controller
+    {
+        public Address2 CollectionType(Address2 address)
+        {
+            return address;
+        }
+
+        public UserWithAddress NestedCollectionType(UserWithAddress user)
+        {
+            return user;
+        }
+
+        public PeopleModel NestedCollectionOfRecursiveTypes(PeopleModel model)
+        {
+            return model;
+        }
+
+        public bool PostCheckBox(bool isValid)
```

Comment:
This doesn't necessarily hit the scenarios we'd want to test - leave this here as is - but could we get one where multiple values are selected:

```C#
public string CheckboxList(IEnumerable<UserPreferences> userPreferences)
{

}

public class UserPreference
{
   int Id { get; set; }
   
   bool Checked { get; set; }
}
```

And the post would look like `[0].Id = Foo&[0].Checked=true&[0].Checked=false&[1].Id=Bar&[1].Checked=false`